network/delegate_packet_proxy.go

```diff
@@ -19,14 +19,14 @@ import (
 	"sync/atomic"
 )
 
-// DelegatePacketProxy is a PacketProxy that forwards calls (like NewSession) to another PacketProxy. To create a
-// DelegatePacketProxy with the default PacketProxy, use NewDelegatePacketProxy. To change the underlying PacketProxy,
-// use SetProxy.
+// DelegatePacketProxy is a [PacketProxy] that forwards calls (like NewSession) to another [PacketProxy]. To create a
```

Comment:
You may want to call out the intent of the Delegate, which is to replace the packetProxy without changing the original reference.

Reply:
added